### DISCUSSION TITLE
Error launching app after successful AoT compilation

### DISCUSSION THREAD

**Discussion by darenm** [unoplatform/uno#11895] [Q&A] [answered]
What are the typical causes of the following runtime error following an AoT compilation:

![image](https://user-images.githubusercontent.com/1674843/229849299-e2ca9d56-0e89-4012-bbcb-d6ccd938bd2a.png)

**Comment by jeromelaban** [comment]
I don't think that the trace logger is supported in the AOT configuration. Are you trying to troubleshoot memory usage ?

**Reply by darenm** [reply]
Not in this specific scenario - perhaps it is an inherited configuration item.

**Reply by jeromelaban** [reply]
Curious. Make sure that https://github.com/unoplatform/Uno.Wasm.Bootstrap/blob/main/doc/features-profiling.md#using-the-memory-profiler is disabled in your build.

**Reply by darenm** [reply]
These are the build settings I am running:

```xml
	<!-- ProfileGuided-AOT -->
	<PropertyGroup Label="Globals" Condition="'$(Configuration)'=='ProfileGuided-AOT'">
		<WasmShellEnableAOTDeduplication>false</WasmShellEnableAOTDeduplication>
		<WasmShellEmccLinkOptimization>true</WasmShellEmccLinkOptimization>
		<WasmShellEmccLinkOptimizationLevel>Maximum</WasmShellEmccLinkOptimizationLevel>
		<WasmShellMonoRuntimeExecutionMode>InterpreterAndAOT</WasmShellMonoRuntimeExecutionMode>
		<!--<WasmShellAOTProfileExcludedMethods>ManifestBasedResourceGroveler;InternalGetSatelliteAssembly;InternalLoad;Dispose;ExecuteCallbackHandlers</WasmShellAOTProfileExcludedMethods>-->
		<WasmShellEnableEmscriptenWindows>false</WasmShellEnableEmscriptenWindows>
		<WasmShellGenerateAOTProfileDebugList>true</WasmShellGenerateAOTProfileDebugList>
		<WasmShellAssembliesExtension>.wlr</WasmShellAssembliesExtension>
		<WasmShellEnableLogProfiler>false</WasmShellEnableLogProfiler>

	</PropertyGroup>
	<PropertyGroup Condition="'$(Configuration)|$(Platform)'=='ProfileGuided-AOT|AnyCPU'">
		<DefineConstants>WASM</DefineConstants>
		<Optimize>true</Optimize>
		<IsUiAutomationMappingEnabled>true</IsUiAutomationMappingEnabled>
	</PropertyGroup>

	<ItemGroup Condition="'$(Configuration)'=='ProfileGuided-AOT'">
		<WasmShellEnableAotProfile Include="aot.profile" />
		<WasmShellExtraEmccFlags Include="-s MAXIMUM_MEMORY=4GB" />

		<WasmShellMonoEnvironment Include="MONO_GC_PARAMS" Value="soft-heap-limit=512m,nursery-size=64m,evacuation-threshold=66,major=marksweep" />
		<WasmShellMonoEnvironment Include="MONO_LOG_LEVEL" Value="debug" />
		<WasmShellMonoEnvironment Include="MONO_LOG_MASK" Value="all" />
	</ItemGroup>
```

**Reply by jeromelaban** [reply]
Then this would likely mean a different issue. Maybe the compilation did not finish properly, or maybe assembly files are not downloaded properly. Take a look at the network tab to see if the content downloaded by the browser makes sense.

**Comment by darenm** [comment, marked as answer]
@jeromelaban Correctly identified this as a content download issue. In order to address issues where customers of our application have firewall restrictions that block the `.clr` extension when downloading web content, we have adopted the Uno configuration option  `WasmShellAssembliesExtension` below.

```xml
<PropertyGroup Label="Globals" Condition="'$(Configuration)'=='Mixed_AOT'">
  <WasmShellEnableAOTDeduplication>false</WasmShellEnableAOTDeduplication>
  <WasmShellEmccLinkOptimization>false</WasmShellEmccLinkOptimization>
  <WasmShellMonoRuntimeExecutionMode>InterpreterAndAOT</WasmShellMonoRuntimeExecutionMode>
  <WasmShellEnableEmscriptenWindows>false</WasmShellEnableEmscriptenWindows>
  <WasmShellAssembliesExtension>.wlr</WasmShellAssembliesExtension>
</PropertyGroup>
```

Unfortunately, when running the solution locally, IISExpress was not serving the content, resulting in the application failing to launch. Removing this setting when running locally resolved the issue.

Full details of the congiuration can be found at [Assemblies obfuscation](https://platform.uno/docs/articles/external/uno.wasm.bootstrap/doc/features-obfsucation.html)
